Pixel Game/Assets/Scripts/Quest.cs

```diff
@@ -13,80 +14,144 @@ public enum QUESTTYPE
         WAIT_TIME,
         DELIVER_ITEM
     }
-    public QUESTTYPE questType;
+    [SerializeField]
+    private QUESTTYPE questType;
 
-    // Refers to Enemyhealth enumtype (to not make double enums)
-    public EnemyHealth.ENEMYTYPE enemyTypeToKill = EnemyHealth.ENEMYTYPE.BEAST;
-
-
-    public string questName;
+    [SerializeField]
+    private string questName;
     [TextArea(2,5)]
-    public string description;
+    [SerializeField]
+    private string description;
+    [SerializeField]
+    private int moneyReward;
+    public int MoneyReward => moneyReward;
+    [SerializeField]
+    private GameObject gameObjectReward;
+    public GameObject GameObjectReward => gameObjectReward;
 
-    public int moneyReward;
-    public GameObject gameObjectReward;
     private PlayerInventory playerInventory;
-
-    public bool isQuestFinished = false;
+    private bool isQuestFinished = false;
+    public bool IsQuestFinished => isQuestFinished;
     // Kill enemies quest
     [Header("Kill enemies quest")]
-    public int amountToKill;
-    public int currentKilled;
-
+    [SerializeField]
+    private string enemyToKill;
+    [SerializeField]
+    private int amountToKill;
+    private int currentKilled;
 
     // Gather items quest
     [Header("Gather items quest")]
-    public GameObject itemToGather;
-    public int amountToCollect;
-    public int currentCollected;
+    [SerializeField]
+    private GameObject itemToGather;
+    [SerializeField]  
+    private int amountToCollect;
+    private int currentCollected;
 
     // Item delivery quest
     [Header("Item delivery quest")]
-    public GameObject itemToDeliver;
-    public GameObject npcDeliveryTarget;
+    [SerializeField]
+    private GameObject itemToDeliver;
+    [SerializeField]
+    private GameObject npcDeliveryTarget;
 
-    void Start()
+    public void Init()
     {
         playerInventory = GameObject.FindWithTag("InventoryManager").GetComponent<PlayerInventory>();
     }
 
+    public bool TryIncrementItemsCollected(string itemName)
+    {
+        if (IsCorrectItem(itemName))
+        {
+            isQuestFinished = ++currentCollected >= amountToCollect;

```

Comment:
What happens here on isQuestFinished?

Reply:
First `currentCollected` is incremented. Then we check if it is `>= amountToCollect`. If it is, we assign true to `isQuestFinished`, otherwise false.